### PR TITLE
Clean up universal flow variables

### LINKED ISSUE #362: Cleanup unused variables
### Description

A number of unused variables from OpenLane 1 continue to exist in the universal flow configuration variables.

### Proposal

They should be removed.

### PR DESCRIPTION
**Pull request by donn**
* Universal Flow Variables
  * Renamed `FP_WELLTAP_CELL` to `WELLTAP_CELL` with translation behavior
  * Renamed `FP_ENDCAP_CELL` to `ENDCAP_CELL` with translation behavior
  * **API Breaks**
    * `CTS_ROOT_BUFFER`, `CTS_CLK_BUFFERS` and `CTS_MAX_CAP` all moved to `OpenROAD.CTS`
    * `IGNORE_DISCONNECTED_MODULES` moved to `Odb.ReportDisconnectedPins`
    * `GPL_CELL_PADDING` moved to `OpenROAD.GlobalPlacement`
    * `DPL_CELL_PADDING` moved to steps that have the rest of the `dpl` variables
    * `GRT_LAYER_ADJUSTMENTS` moved to steps that have the rest of the routing layer variables
    * Moved `GRT_OBS` to `Odb.AddRoutingObstructions` as a deprecated name for `ROUTING_OBSTRUCTIONS`
    * Removed `FP_CONTEXT_DEF`, `FP_CONTEXT_LEF`, and `FP_PADFRAME_CFG`: To be implemented
    * Removed `LVS_INSERT_POWER_PINS`, `RUN_CVC`, `LEC_ENABLE`, `CHECK_ASSIGN_STATEMENTS`

---

Resolves #362

### CODE REVIEW

File: openlane/config/flow.py [hunk 1]
```diff
@@ -344,45 +344,19 @@ def _prefix_to_wildcard(prefixes_raw: Union[str, Sequence[str]]):
...
         str,
         "Defines the cell used for tap insertion.",
         pdk=True,
+        deprecated_names=["FP_WELLTAP_CELL"],
     ),
     Variable(
-        "FP_ENDCAP_CELL",
+        "ENDCAP_CELL",
         str,
         "Defines so-called 'end-cap' cells- decap cells placed at either sides of a design.",
```
Comment (kareefardi) [inline openlane/config/flow.py:357]: might as well remove "so-called"

File: openlane/config/removals.py [hunk 2]
```diff
@@ -36,4 +36,10 @@
     "MAGIC_GDS_ALLOW_ABSTRACT": "Bad practice. If an abstract view is needed, a GDS file can be generated from the abstract LEF file.",
     "PLACE_SITE_HEIGHT": "Now automatically extracted from PLACE_SITE.",
     "PLACE_SITE_WIDTH": "Now automatically extracted from PLACE_SITE.",
+    "LEC_ENABLE": "Complex, unscalable, and largely reundant.",
```
Comment (kareefardi) [inline openlane/config/removals.py:39]: Shouldn't `LEC_ENABLE` a deprecated variable for `RUN_EQY`

File: openlane/steps/odb.py [hunk 3]
```diff
@@ -278,6 +278,16 @@ class ReportDisconnectedPins(OdbpyStep):
     id = "Odb.ReportDisconnectedPins"
     name = "Report Disconnected Pins"
 
+    config_vars = OdbpyStep.config_vars + [
+        # Why is this a PDK variable again
```
Comment (kareefardi) [inline openlane/steps/odb.py:282]: Because it references PDK cells so it is PDK dependent.

**Review by kareefardi — CHANGES_REQUESTED** [review]
See above comments. ~should have added them as part of the review~

File: openlane/config/flow.py (hunk 1 above)
Comment (donn) [reply openlane/config/flow.py:357]: we've been over this

File: openlane/config/removals.py (hunk 2 above)
Comment (donn) [reply openlane/config/removals.py:39]: No, it behaves fairly differently

File: openlane/steps/odb.py (hunk 3 above)
Comment (donn) [reply openlane/steps/odb.py:282]: Fair enough

**Review by kareefardi — APPROVED** [review]

**Pull request merged**